yb-voyager/cmd/importData.go

```diff
@@ -193,19 +193,48 @@ func getYBServers() []*tgtdb.Target {
 		log.Infof("Target DB nodes: %s", strings.Join(hostPorts, ","))
 	}
 
-	if parallelImportJobs == -1 {
-		parallelImportJobs = len(targets)
-	}
-
 	if loadBalancerUsed { // if load balancer is used no need to check direct connectivity
 		utils.PrintAndLog(LB_WARN_MSG)
+		parallelImportJobs = 2 * len(targets)
+		utils.PrintAndLog("Using %d parallel jobs by default. Use --parallel-jobs to specify a custom value", parallelImportJobs)
 		targets = []*tgtdb.Target{&target}
 	} else {
 		testYbServers(targets)
 	}
 	return targets
 }
 
+func fetchDefaultParllelJobs(targets []*tgtdb.Target) int {
+	coreQueries := []string{"CREATE TEMP TABLE yb_voyager_cores(num_cores int);",
+		"COPY yb_voyager_cores(num_cores) FROM PROGRAM 'grep processor /proc/cpuinfo|wc -l';",
+		"SELECT num_cores FROM yb_voyager_cores;"}
+	totalCores := 0
+	targetCores := 0
+	for _, target := range targets {
+		conn, err := pgx.Connect(context.Background(), target.Uri)
+		if err != nil {
+			utils.ErrExit("Unable to reach target while querying cores: %v", err)
```

Comment:
We must not fail+exit data import when we cannot determine core count. Instead, just log a WARNING msg and use the default `len(targets) * 2`.